### PR TITLE
Relax the strict version pins for lmdbm, mpire, orjson, and pennylane

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
This is a test to see if removing the version pin on `orjson` breaks things (also testing the others added in #1794).

Edit: anddddd success 🙏

- [X] I have added the tests to cover my changes.
- [X] I have updated the documentation and CHANGELOG accordingly.
- [X] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by Andrew-S-Rosen** [comment]
@kessler-frost: Looks like this passed. I'm going to update the CHANGELOG. Feel free to propose any changes from here.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Operations
 
+- Remove version pin on `lmdbm`, `mpire`, `orjson`, and `pennylane`
```
Comment (kessler-frost) [inline CHANGELOG.md:17]: Might wanna move this to the `Changed` section.

**Review by kessler-frost — CHANGES_REQUESTED** [review]
Thanks @Andrew-S-Rosen ! Just a minor comment.

**Comment by Andrew-S-Rosen** [comment]
@kessler-frost: Thanks! I agree about the CHANGELOG category change. All fixed!

**Review by kessler-frost — APPROVED** [review]

**Pull request merged**